controllers/tasks.py

```diff
@@ -0,0 +1,35 @@
+import webapp2
+from utils.decorators import user_required
+from utils import render_template, render_json
+from models.task import Task
+from models.project import Project	
+		
+class TaskHandler(webapp2.RequestHandler):
+	
+	@user_required
+	def get(self):
+		task_token = self.request.get('key')
+		task = Task.find_with_key(task_token)
+		params = {
+			'task': task
+		}
+		render_template(self, 'task.html', params)
+		
+	def post(self):
+		task.completed == true
+		
+class TaskCreateHandler(webapp2.RequestHandler):
+	
+	@user_required
+	def get(self):
+		render_template(self, 'add_task.html')
+		
+	def post(self):
```

Comment:
This isn't quite right; take a look at how the code is set up to [add a project to a course](https://github.com/AAAStrix/ctrl-class/blob/master/controllers/projects.py#L53-L70) for a better idea of how adding a task to a project will work.  The code will be really similar, just swapping the models.

Note that when you're adding the task to the project, you can skip the bit about adding a project to the user.